packages/flutter_tools/lib/src/commands/run_mojo.dart

```diff
@@ -87,20 +88,28 @@ class RunMojoCommand extends FlutterCommand {
     final String command = useDevtools ? _getDevtoolsPath() : _getMojoShellPath();
     args.add(command);
 
+    BuildConfiguration config = _getCurrentHostConfig();
+
     if (argResults['android']) {
       args.add('--android');
-      final String cloudStorageBaseUrl = ArtifactStore.getCloudStorageBaseUrl('shell', 'android-arm');
       final String appPath = _makePathAbsolute(bundlePath);
       final String appName = path.basename(appPath);
       final String appDir = path.dirname(appPath);
       args.add('http://app/$appName');
       args.add('--map-origin=http://app/=$appDir');
-      args.add('--map-origin=http://flutter/=$cloudStorageBaseUrl');
-      args.add('--url-mappings=mojo:flutter=http://flutter/flutter.mojo');
+      if (config == null || config.type == BuildType.prebuilt) {
+        final String cloudStorageBaseUrl = ArtifactStore.getCloudStorageBaseUrl('shell', 'android-arm');
+        args.add('--map-origin=http://flutter/=$cloudStorageBaseUrl');
+        args.add('--url-mappings=mojo:flutter=http://flutter/flutter.mojo');
+      } else {
+        String localPath = path.join(config.buildDir, 'flutter.mojo');
+        File flutterMojo = new File(localPath);
+        flutterMojo.copySync(path.join(appDir, 'flutter.mojo'));
```

Comment:
It looks like the devtools package does this check to see if it should map something through the web server:

https://github.com/domokit/mojo/blob/master/mojo/devtools/common/devtoolslib/shell_arguments.py#L24

which just checks if it can parse a 'scheme' out.  Seems odd, but to get that logic to detect a filepath as you want you probably want to say '/foo/bar/baz' rather than 'file:///foo/bar/baz'.  If that doesn't work ask ppi how to get this configured